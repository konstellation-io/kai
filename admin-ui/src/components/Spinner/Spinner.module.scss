.contentBox {
  position: relative;
  width: 100%;
  height: 100%;
}

.container {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.sun {
  width: 37%;
  height: 37%;
  border-radius: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.planet_container {
  border-radius: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.planet {
  border-radius: 50%;
  position: absolute;
  top: 0;
  left: 50%;
  transform: translate(-50%, -50%);
}

.orbit {
  border-radius: 50%;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  border: solid black 1px;
  opacity: 0.4;
}

.planet_container.p1 {
  width: 60%;
  height: 60%;
  animation: spin 2.5s linear infinite;
}
.planet.p1 {
  width: 15%;
  height: 15%;
}
.orbit.p1 {
  width: 60%;
  height: 60%;
}

.planet_container.p2 {
  width: 100%;
  height: 100%;
  animation: spin-reverse 3s linear infinite;
}
.planet.p2 {
  width: 6%;
  height: 6%;
}
.orbit.p2 {
  width: 100%;
  height: 100%;
  opacity: 0.2;
}

.planet_container.p3 {
  width: 100%;
  height: 100%;
  animation: spin 8s linear infinite;
}
.planet.p3 {
  width: 6%;
  height: 6%;
}

.planet_container.p4 {
  width: 200%;
  height: 200%;
  animation: spin-reverse 0.8s linear infinite;
}
.planet.p4 {
  width: 20%;
  height: 20%;
}

.planet_container.p5 {
  width: 80%;
  height: 80%;
  animation: spin-reverse 6s linear infinite;
}
.planet.p5 {
  width: 7%;
  height: 7%;
}
.orbit.p5 {
  width: 80%;
  height: 80%;
  opacity: 0.3;
}

@keyframes spin {
  0% {
    transform: translate(-50%, -50%) rotate(0deg);
  }
  100% {
    transform: translate(-50%, -50%) rotate(360deg);
  }
}
@keyframes spin-reverse {
  0% {
    transform: translate(-50%, -50%) rotate(360deg);
  }
  100% {
    transform: translate(-50%, -50%) rotate(0deg);
  }
}
