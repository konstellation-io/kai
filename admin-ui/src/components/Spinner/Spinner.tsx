import React from 'react';
import styles from './Spinner.module.scss';

type Props = {
  size?: number;
  color?: string;
};

function Spinner({ size = 40, color = 'white' }: Props) {
  return (
    <div className={styles.contentBox} data-testid="spinner">
      <div className={styles.container} style={{ width: size, height: size }}>
        <div className={styles.sun} style={{ backgroundColor: color }} />
        <div
          className={`${styles.orbit} ${styles.p1}`}
          style={{ borderColor: color }}
        />
        <div className={`${styles.planet_container} ${styles.p1}`}>
          <div
            className={`${styles.planet} ${styles.p1}`}
            style={{ backgroundColor: color }}
          >
            <div
              className={`${styles.planet_container} ${styles.p4}`}
              data-testid="spinner-planet"
            >
              <div
                className={`${styles.planet} ${styles.p4}`}
                style={{ backgroundColor: color }}
              />
            </div>
          </div>
        </div>
        <div
          className={`${styles.orbit} ${styles.p2}`}
          style={{ borderColor: color }}
        />
        <div className={`${styles.planet_container} ${styles.p2}`}>
          <div
            className={`${styles.planet} ${styles.p2}`}
            style={{ backgroundColor: color }}
          />
        </div>
        <div className={`${styles.planet_container} ${styles.p3}`}>
          <div
            className={`${styles.planet} ${styles.p3}`}
            style={{ backgroundColor: color }}
          />
        </div>
        <div
          className={`${styles.orbit} ${styles.p5}`}
          style={{ borderColor: color }}
        />
        <div className={`${styles.planet_container} ${styles.p5}`}>
          <div
            className={`${styles.planet} ${styles.p5}`}
            style={{ backgroundColor: color }}
          />
        </div>
      </div>
    </div>
  );
}

export default Spinner;
