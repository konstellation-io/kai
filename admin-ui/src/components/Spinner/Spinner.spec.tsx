import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { getByTestId } from '@testing-library/dom';
import Spinner from './Spinner';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

it('Render Spinner without crashing', () => {
  const { container } = render(<Spinner />);
  expect(container).toMatchSnapshot();

  expect(getByTestId(container, 'spinner-planet')).toBeInTheDocument();
});
