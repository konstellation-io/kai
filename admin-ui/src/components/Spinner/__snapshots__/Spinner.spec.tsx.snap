// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Spinner without crashing 1`] = `
<div>
  <div
    class="contentBox"
    data-testid="spinner"
  >
    <div
      class="container"
      style="width: 40px; height: 40px;"
    >
      <div
        class="sun"
        style="background-color: white;"
      />
      <div
        class="orbit p1"
        style="border-color: white;"
      />
      <div
        class="planet_container p1"
      >
        <div
          class="planet p1"
          style="background-color: white;"
        >
          <div
            class="planet_container p4"
            data-testid="spinner-planet"
          >
            <div
              class="planet p4"
              style="background-color: white;"
            />
          </div>
        </div>
      </div>
      <div
        class="orbit p2"
        style="border-color: white;"
      />
      <div
        class="planet_container p2"
      >
        <div
          class="planet p2"
          style="background-color: white;"
        />
      </div>
      <div
        class="planet_container p3"
      >
        <div
          class="planet p3"
          style="background-color: white;"
        />
      </div>
      <div
        class="orbit p5"
        style="border-color: white;"
      />
      <div
        class="planet_container p5"
      >
        <div
          class="planet p5"
          style="background-color: white;"
        />
      </div>
    </div>
  </div>
</div>
`;
